### PR TITLE
feat: upgrade Azure Functions to v4 in dashboard related samples

### PR DESCRIPTION
**Pull request by huimiu**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/26023634
- Upgrade to [Azure Function V4 programming model](https://techcommunity.microsoft.com/t5/apps-on-azure-blog/azure-functions-node-js-v4-programming-model-is-generally/ba-p/3929217).
- Remove function binding.
- Fix some typo issues.

### CODE REVIEW

File: developer-assist-dashboard/api/package.json [hunk 1]
```diff
@@ -3,15 +3,15 @@
   "version": "1.0.0",
   "scripts": {
     "dev:teamsfx": "env-cmd --silent -f .localConfigs npm run dev",
-    "dev": "func start --typescript --language-worker=\"--inspect=9229\" --port \"7071\" --cors \"*\"",
+    "dev": "nodemon --exec func start --typescript --language-worker=\"--inspect=9229\" --port \"7071\" --cors \"*\"",
```
Comment (hund030) [inline developer-assist-dashboard/api/package.json:6]: Why do we need nodemon? It seems that it works without nodemon.
https://github.com/OfficeDev/TeamsFx/blob/dev/templates/ts/sso-tab-with-obo-flow/api/package.json#L10

File: developer-assist-dashboard/package.json [hunk 2]
```diff
@@ -19,12 +19,11 @@
     "react-scripts": "^5.0.1"
   },
   "devDependencies": {
-    "@types/node": "^14.0.0",
+    "@types/node": "^18.16.1",
```
Comment (hund030) [inline developer-assist-dashboard/package.json:22]: Also update the engines version in Line 5?

File: intelligent-data-chart-generator/api/package.json [hunk 3]
```diff
@@ -3,20 +3,22 @@
   "version": "1.0.0",
   "scripts": {
     "dev:teamsfx": "env-cmd --silent -f .localConfigs npm run dev",
-    "dev": "func start --typescript --language-worker=\"--inspect=9229\" --port \"7071\" --cors \"*\"",
+    "dev": "nodemon --exec func start --typescript --language-worker=\"--inspect=9229\" --port \"7071\" --cors \"*\"",
```
Comment (hund030) [inline intelligent-data-chart-generator/api/package.json:6]: Also here.

File: developer-assist-dashboard/api/package.json (hunk 1 above)
Comment (huimiu) [reply developer-assist-dashboard/api/package.json:6]: I've noticed that sometimes the Azure Functions' code doesn't hot-reload during local debugging. There's a bug tracking this issue. Adding this is the current workaround.

File: developer-assist-dashboard/package.json (hunk 2 above)
Comment (huimiu) [reply developer-assist-dashboard/package.json:22]: Updated

**Review by hund030 — DISMISSED** [review]

File: developer-assist-dashboard/api/package.json (hunk 1 above)
Comment (huimiu) [reply developer-assist-dashboard/api/package.json]: The hot reload issue has been resolved.

**Review by hund030 — APPROVED** [review]

**Pull request merged**